tests/src/system/basic/WskActionTests.scala

```diff
@@ -0,0 +1,268 @@
+            val activation = wsk.parseJsonString(run.stdout).convertTo[CliActivation]
+
+            withClue(s"check failed for activation: $activation") {
+                val wordCount = testString.split(" ").length
+                activation.response.result.get shouldBe JsObject("binaryCount" -> s"${wordCount.toBinaryString} (base 2)".toJson)
+            }
+    }
+
+    it should "blocking invoke an asynchronous action" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "helloAsync"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/helloAsync.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson), blocking = true)
+            val activation = wsk.parseJsonString(run.stdout).convertTo[CliActivation]
+
+            withClue(s"check failed for activation: $activation") {
+                activation.response.status shouldBe "success"
+                activation.response.result shouldBe Some(testResult)
+                activation.logs shouldBe None
+            }
+    }
+
+    it should "return the value of the first synchronous whisk.done()" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "helloSyncDoneTwice"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("helloSyncDoneTwice.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.response.result shouldBe Some(testResult)
+                    activation.logs.get.mkString(" ") should include(testString)
+            }
+    }
+
+    it should "return the value of the first asynchronous whisk.done()" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "helloSyncDoneTwice"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("helloAsyncDoneTwice.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.response.result shouldBe Some(testResult)
+                    activation.logs.get.mkString(" ") should include(testString)
+            }
+    }
+
+    it should "reject an invoke with the wrong parameters set" in {
+        val payload = "bob"
+        val rr = wsk.cli(Seq("action", "invoke", "/whisk.system/samples/helloWorld", payload), expectedExitCode = TestUtils.ANY_ERROR_EXIT)
```

Comment:
After the catalog is removed, the action "/whisk.system/samples/helloWorld" does not exit by default. Please create the package samples and the action helloWorld before this line with assetHelper.withCleaner.
You can use wskprops as the whisk properties.